homeassistant/components/switch/flux.py

```diff
@@ -125,6 +125,8 @@ def setup_platform(hass, config, add_entities, discovery_info=None):
     mode = config.get(CONF_MODE)
     interval = config.get(CONF_INTERVAL)
     transition = config.get(ATTR_TRANSITION)
+    if transition > interval:
```

Comment:
Move this to a validator function and use that in the config schema.

Reply:
I am not sure how to do that so I have removed this unrelated improvement.

Reply:
Ok. Something like this:
```py
def is_valid_transition(config):
    """Validate transition compared to interval."""
    transition = config[CONF_TRANSITION]
    interval = config[CONF_INTERVAL]
    if transition > interval:
        transition = interval
        config[CONF_TRANSITION] = transition
    return config

PLATFORM_SCHEMA = vol.Schema(vol.All(PLATFORM_SCHEMA.extend({
    ...
    vol.Optional(CONF_INTERVAL, default=30): cv.positive_int,
    vol.Optional(CONF_TRANSITION, default=30): VALID_TRANSITION,
}), is_valid_transition))
```